test/multiverse/suites/agent_only/agent_run_id_handling_test.rb

```diff
@@ -33,9 +33,15 @@ def test_handles_non_numeric_agent_run_id_on_slow_sql_traces
   end
 
   def simulate_slow_sql_trace
-    in_transaction do
-      agent.sql_sampler.notice_sql('select * from test', 'Database/test/select', nil, 1.5)
+    with_config(:'slow_sql.explain_threshold' => 0.0) do
+      in_transaction do
+        segment = NewRelic::Agent::Tracer.start_datastore_segment
+        # metric name and elapsed args are ignored
+        NewRelic::Agent::Datastores.notice_sql('select * from test', 'Database/test/select', 1.5)
+        segment.finish
+      end
     end
```

Comment:
There's two different strategies for replacing the sql_sampler.notice_sql method in tests. This example above and: 

```
statement = NewRelic::Agent::Database::Statement.new('select * from test')
    agent.sql_sampler.notice_sql_statement(statement, 'Database/test/select', 1.5)
```

I tried to use the first strategy when there was clearly a transaction running in the test and the second strategy when transactions seem less important.

Reply:
nice!